logstash-core-plugin-api/logstash-core-plugin-api.gemspec

```diff
@@ -11,8 +11,10 @@ if File.exist?(project_versions_yaml_path)
   #
   # we ignore the copy in git and we overwrite an existing file
   # each time we build the logstash-core gem
-  original_lines = IO.readlines(project_versions_yaml_path)
-  original_lines << ""
+  original_lines = IO.read(project_versions_yaml_path)
+  # introduce the version qualifier (e.g. beta1, rc1) into the copied yml so it's displayed by Logstash
+  original_lines.gsub!(/^logstash: (.+?)\n/, "logstash: \\1-#{ENV['VERSION_QUALIFIER']}\n") if ENV['VERSION_QUALIFIER']
```

Comment:
I kind of liked the `readlines` approach. 
```
original_lines = IO.readlines(project_versions_yaml_path)
if ENV['VERSION_QUALIFIER']
  original_lines.map! do |line|
    if line.start_with?("logstash: ")
      line.chomp + "-#{ENV['VERSION_QUALIFIER']}\n"
    else
      line
    end
  end
end
```
Doing the regex sub seems like we could potentially get in to trouble with whitespace or newline weirdness.

Reply:
Refactored to keep the readlines! thanks